hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "Power Manager HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies the functionality of the OpenTitan power manager.
+
+## Features
+
+- Cold boot, low power entry / exit and reset support.
+- 2 different low power modes.
+- Software initiated low power entry and hardware requested low power exit.
+- Peripheral reset requests
+- Low power abort and low power fall-through support.
+
+## Description
+
+The power manager sequeces power, clocks, and reset resources of the design through cold boot, low power entry/exit and reset scenarios.
+
+Cold boot, also known as POR (power on reset) is the first reset state of the design.
+The power manager sequences the design from a freshly reset state to an active state where software can be initialized.
+
+Low power entry is the process in which the device enters one of two low power modes (sleep or deep sleep).
+Low power exit is the process in which the device exits low power mode and returns to active state.
+Low power entry is always initiated by software, while low power exit is always initiated by a previously setup hardware event such as pins or internal timers.
+The power manager processes the software and hardware requests to perform the appropriate actions.
+
+Reset scenarios refer to non-por events that cause the device to reboot.
+There are various stimuls that can cause such a reset, ranging from exteranl user input to watchdog.
+The power manager processes the reset request and brings the device to an appropriate state.
+
+
+# Theory Of Operation
+
+The power manager controls power domain(s) turn on / turn off, the root resets in conjunction with the reset manager, the root clock enables in conjunction with AST / clock manager and various power up activities such as OTP sensing and life cycle initiation before allowing software to execute.
+
+In the future, if there is a need to perform ROM hashing as an integrity check, that will also be managed by the power manager.
+
+## Block Diagram
+
+See the below high level block diagram that illustrates the connections between the power manager and various system components.
+Blocks outlined with a solid magenta line are always on; while blocks outlined with a dash magenta line are a mix of always-on and non-always-on components.
+
+![Power Manager Connectivity Diagram](pwrmgr_connectivity.svg)
+
+
+## Overall Sequencing
+
+The power manager primarily is composed of two state machines.
+One operates on the always-on slow clock (this clock is always running and usually measured in KHz) and is responsible for turning faster clocks on and off and managing the power domains.
+The other operates on a normal fixed clock (usually measured in MHz) and is responsible for everything else in the power sequence.
+
+The following diagram breaks down the general functionality of both.
+
+
+![Power Manager FSMs](pwrmgr_fsms.svg)
```

Comment:
i added some text to explain the colors, and yes i do think I was supposed to have bolded Active and not dis_clks.
Maybe I accidentally copy / pasted and forgot.  Let me just remove it for now.

Reply:
Oh, sorry, I meant the colours in the text (so "ack_pwrup" is green, "!req_pwrup" is orange). Is this just to show that the signal comes from the other FSM?

Assuming that "!foo" means "foo is not asserted", it might be sensible to change it to "foo = 0", otherwise you collide with the send/receive notation that's usually used for communicating state machines (for example, see the Wikipedia article on [CSP](https://en.wikipedia.org/wiki/Communicating_sequential_processes)): I originally thought that this is what you meant.

I think I still don't understand the bold ellipses and signals in the FSM diagram either.

And (nit) it would still be nice if the dashed line was extended to the bottom or vertically centred (I'm listing it again so that everything appears in the last comment). Obviously, this isn't massively important!

Reply:
oops! i actually fixed some of this but did not upload the diagram. (also forgot to extend the dashed line).
I'll do that soon and let you know.